mlir-sycl/lib/Dialect/SYCL/IR/SYCLOps.cpp

```diff
@@ -83,10 +83,12 @@ bool SYCLAddrSpaceCastOp::areCastCompatible(TypeRange inputs,
   if (input.getLayout() != output.getLayout())
     return false;
 
-  unsigned int inputMS = input.getMemorySpaceAsInt();
-  unsigned int outputMS = output.getMemorySpaceAsInt();
-  return ((inputMS == genericAddressSpace) !=
-          (outputMS == genericAddressSpace));
+  auto inputMS = dyn_cast_or_null<AccessAddrSpaceAttr>(input.getMemorySpace());
+  auto outputMS =
+      dyn_cast_or_null<AccessAddrSpaceAttr>(output.getMemorySpace());
+  return (inputMS && outputMS &&
+          (inputMS.getValue() == AccessAddrSpace::GenericAccess) !=
+              (outputMS.getValue() == AccessAddrSpace::GenericAccess));
```

Comment:
This is the only place we're using these attributes for now as a proof of concept. We're not generating this operation in `cgeist`, so no change is needed there.